Platformio/HAL/HardwareAbstract.cpp

```diff
@@ -23,4 +23,26 @@ std::optional<HardwareAbstract::batteryStatus> HardwareAbstract::getBatteryStatu
 
 void HardwareAbstract::onBatteryChange(std::function<void(HardwareAbstract::batteryStatus)> onBatteryStatusChangeHandler){
   mBatteryNotification.onNotify(std::move(onBatteryStatusChangeHandler));
-}
\ No newline at end of file
+}
+
+void HardwareAbstract::onStartWifiScan(std::function<void()> cb_func){
+  this->wifi_scan_start_cb.push_back(cb_func);
+}
+
+void HardwareAbstract::onWifiScanDone(std::function<void(std::shared_ptr<std::vector<WifiInfo>>)> cb_func){
+  this->wifi_scan_done_cb.push_back(cb_func);
```

Comment:
using notification class 
wifi_scan_done.onNotify(std::move(cb_func)); 


The Notification class keeps the vector of callbacks internally :)